echidna/output/store.py

```diff
@@ -69,26 +132,72 @@ def load(file_path):
+                               spectra.SpectraConfig(parameters))
+        spec._raw_events = file_.attrs["raw_events"]
+        try:
+            spec._bipo = file_.attrs["bipo"]
+        except:
+            spec._bipo = 0
+        style_dict = file_.attrs["style"]
+        if len(style_dict) > 0:
+            spec._style = string_to_dict(style_dict)
+        rois_dict = file_.attrs["rois"]
+        if len(rois_dict) > 0:
+            spec._rois = string_to_dict(rois_dict)
+        # else the default values of Spectra __init__ are kept
+        spec._data = file_["data"].value
+    return spec
+
+
+def load_old(file_path):
+    """ Load a spectra from file_path.
+    Args:
+      file_path (string): Location to save to.
+    Returns:
+      Loaded spectra (:class:`spectra.Spectra`).
     """
     with h5py.File(file_path, "r") as file_:
-        spectra = echidna.core.spectra.Spectra(file_.attrs["name"],
-                                               file_.attrs["num_decays"])
-        spectra._energy_low = file_.attrs["energy_low"]
-        spectra._energy_high = file_.attrs["energy_high"]
-        spectra._energy_bins = file_.attrs["energy_bins"]
-        spectra._energy_width = file_.attrs["energy_width"]
-        spectra._radial_low = file_.attrs["radial_low"]
-        spectra._radial_high = file_.attrs["radial_high"]
-        spectra._radial_bins = file_.attrs["radial_bins"]
-        spectra._radial_width = file_.attrs["radial_width"]
-        spectra._time_low = file_.attrs["time_low"]
-        spectra._time_high = file_.attrs["time_high"]
-        spectra._time_bins = file_.attrs["time_bins"]
-        spectra._time_width = file_.attrs["time_width"]
-
-        spectra._data = file_["data"].value
-    return spectra
+        parameters = collections.OrderedDict()
+        parameters["energy"] = \
+            spectra.SpectraParameter("energy",
+                                     file_.attrs["energy_low"],
+                                     file_.attrs["energy_high"],
+                                     file_.attrs["energy_bins"])
+        parameters["radial"] = \
+            spectra.SpectraParameter("radial",
+                                     file_.attrs["radial_low"],
+                                     file_.attrs["radial_high"],
+                                     file_.attrs["radial_bins"])
+        parameters["time"] = spectra.SpectraParameter("time",
+                                                      file_.attrs["time_low"],
+                                                      file_.attrs["time_high"],
+                                                      file_.attrs["time_bins"])
+        spectra_config = spectra.SpectraConfig(parameters)
+        spec = spectra.Spectra(file_.attrs["name"],
+                               file_.attrs["num_decays"],
+                               spectra_config)
+        # The following may not be present in old, old hdf5s
+        if file_.attrs["raw_events"]:
```

Comment:
I tested this functionality as well, when comparing with old spectra. First this is really cool! But I've found it does still raise `KeyError` when these attributes don't exist in the HDF5. You could change it to something like this:

```
        try:
            spec._raw_events = file_.attrs["raw_events"]
        except KeyError as detail:  # unable to locate attribute, skip
            print detail
            print " --> skipping!"
        try:
            spec._style = file_.attrs["style"]
        except KeyError as detail:  # unable to locate attribute, skip
            print detail
            print " --> skipping!"
        try:
            spec._rois = file_.attrs["rois"]
        except KeyError as detail:  # unable to locate attribute, skip
            print detail
            print " --> skipping!"
```

Which I admit is not as nice, but I can't really think how else to do it.